docs/concepts/controller-manager.md

```diff
@@ -54,6 +54,22 @@ The component configuration of the Gardener Controller Manager offers to configu
 * `staleGracePeriodDays`: Don't compute auto-delete timestamps for stale `Project`s that are unused for only less than `staleGracePeriodDays`. This is to not unnecessarily make people/end-users nervous "just because" they haven't actively used their `Project` for a given amount of time. When you change this value then already assigned auto-delete timestamps may be removed again if the new grace period is not yet exceeded.
 * `staleExpirationTimeDays`: Expiration time after which stale `Project`s are finally auto-deleted (after `.status.staleSinceTimestamp`). If this value is changed and an auto-delete timestamp got already assigned to the projects then the new value will only take effect if it's increased. Hence, decreasing the `staleExpirationTimeDays` will not decrease already assigned auto-delete timestamps.
 
+### Event Controller
+
+With the Gardener Event Controller you can prolong the lifespan of shoot cluster maintenance and hibernation events.
+This is an optional controller which will become active once you provide the below mentioned configuration.
+
+All events in K8s are deleted after a configurable [kube-apiserver argument](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/) called `--event-ttl` (defaulting to 1 hour).
+The need to prolong the time-to-live for certain events frequently arises when debugging life systems.
+This is especially true for events impacting the lifecycle of customers clusters directly, namely maintenance and hibernation events concerning shoot clusters.
+This controller leaves the above mentioned events untouched while deleting all other events after a configured time.
+In order to activate it provide the following configuration:
+
+* `ttlShortLivedEvents`: When an event reaches this time-to-live it gets deleted unless it is a maintenance or hibernation event.
```

Comment:
Hmm, why is it limited to hibernation/maintenance events (instead of simply capturing/keeping all shoot-related events like reconciliations and potential future other events)?

Reply:
I think we are afraid that events could `pollute` etcd so it runs out of space. What if someone introduces a very frequently recorded event that is now kept for 14days?
An example for this was this event which we did not want to persist. In this case we just deleted it: https://github.com/gardener/gardener/pull/2649/files/cbdeb83c7c2ad487fef4eae4878dd0a6081e16a2#diff-a6ee7351b6be637d35413a1a0f49bedcL224

Reply:
End-users are not allowed to create events, so they can only come from operators/components. I'm thinking about the use-case where an extension controller wants to submit events, too. With the current whitelist this would not be easily possible (because it would be garbage-collected too fast), and the extensions would have to implement a similar event controller to keep their events.
So, I don't think there is a risk of pollution at this point in time as event creation is not exposed to end-users.

Reply:
As I discussed this also with @timebertt WDYT?

Reply:
Good point. I think, it makes sense to keep all events referring to `Shoot` objects instead of only ones with whitelisted `Reasons`.

Reply:
I looked at common events with ` kubectl get events --all-namespaces --field-selector "involvedObject.kind==Shoot" --sort-by=reason`. Overwhelmingly they are `SchedulingSuccessful`, `ScheduledNextSync`, `Reconciling` and `Reconciled` and `ReoncileError`. Then some are `deleting` and `deleted`.
1. I do not have live data but it seems these events can add up quickly if we save all of them of the last for example 14 days.
2. Do these really need to be events? Is ScheduledNextSync important enough to be recorded as an event?

Reply:
We also removed this event because it did not seem to be helpful: https://github.com/gardener/gardener/pull/2649/files#diff-a6ee7351b6be637d35413a1a0f49bedcL226

Reply:
Thanks for checking this. I think it would be helpful to know when a reconciliation/deletion operations ran, so we can keep `Reconciling` an `Deleting`, for example, and remove all others. Together with the to-be-introduced operation ID (that should become part of the event message as well) we can later easily find the correct logs, so it's not necessary to send events for the result of the operation (e.g., `ReconcileError` or `DeletionSuccesful`). At least we could start like this and gain some experience. If we see that we the data volume is not growing to large we can still get those events back.

Reply:
Now all Shoot-related events will be persisted. I changed the code and documentation.

Reply:
I tried to adapt the above suggestions regarding replacing events with logs as best as I could in the current commit.

Reply:
Could you recheck if the events are now fired correctly / logs are written as expected? I left `shoot_control_migrate.go` untouched as it is a new component and we can gather some experience with that first. Also it should not happen frequently I assume.